pgq-processors/ext/pgq/lib/pgq/worker.rb

```diff
@@ -2,10 +2,12 @@ class Pgq::Worker
   attr_reader :logger, :class, :sleep_time
 
   cattr_accessor :interrupted do false end
+  cattr_accessor :interrupted_with_exception do false end
   cattr_accessor :logger
 
-  def self.shutdown!(signal)
-     self.interrupted = signal
+  def self.shutdown!(signal, with_exception = false)
+    self.interrupted = signal
```

Comment:
@senid231 maybe simplify it and leave 1 argument but check if exception inside shutdown! and set interrupted_with_exception as private

Reply:
done